service/controller/v14/resource/endpoint/current_test.go

```diff
@@ -216,10 +220,10 @@ func Test_Resource_Endpoint_GetCurrentState(t *testing.T) {
 		}
 		result, err := newResource.GetCurrentState(resourcecanceledcontext.NewContext(context.TODO(), make(chan struct{})), tc.Obj)
 		if err != nil {
-			t.Fatal("case", i+1, "expected", nil, "got", err)
+			t.Fatal("case", i, "expected", nil, "got", err)
 		}
 		if !reflect.DeepEqual(tc.ExpectedEndpoints, result) {
-			t.Fatalf("case %d expected %#v got %#v", i+1, tc.ExpectedEndpoints, result)
+			t.Fatalf("case %d expected %#v got %#v", i, tc.ExpectedEndpoints, result)
```

Comment:
The tests are not so cool. Just fixed some of the most annoying things I found when scanning through.